backend/tests/v5/unit/processors/teamspaces/projects/models/commons/tickets.groups.test.js

```diff
@@ -37,244 +38,342 @@ const ScenesModel = require(`${src}/models/scenes`);
+						const projection = { metadata: { $elemMatch:
+							{ $or: externalIdKeys.map((n) => ({ key: n })) } } };
+						expect(MetaModel.getMetadataByQuery).toHaveBeenCalledWith(teamspace, obj.container,
+							query, projection);
+
+						return metadata.length
+							? { ...obj, _ids: undefined, [externalIdName]: metadata.map((m) => m.metadata[0].value) }
+							: obj;
+					});
+					return { ...group, objects: convertedObjects };
+				});
+
+				expect(GroupsModel.addGroups).toHaveBeenCalledTimes(1);
+				expect(GroupsModel.addGroups).toHaveBeenCalledWith(teamspace, project, container,
+					ticket, convertedGroups);
+			});
 		});
+	});
+};
 
-		test('[Smart group] should return the group found with the query results (idmap not found)', async () => {
-			const groupData = { ...generateRandomObject(), rules: [generateRandomObject()] };
-			GroupsModel.getGroupById.mockResolvedValueOnce(groupData);
-
-			const metaRes = times(10, () => ({ parents: times(2, () => generateRandomString()) }));
-			const rejected = times(10, () => ({ parents: times(2, () => generateRandomString()) }));
-			MetaModel.getMetadataByRules.mockResolvedValueOnce({ matched: metaRes, unwanted: rejected });
-
-			const nodeRes = times(10, () => ({ _id: generateRandomString() }));
-			ScenesModel.getNodesBySharedIds.mockResolvedValueOnce(nodeRes);
-			ScenesModel.getNodesBySharedIds.mockResolvedValueOnce(nodeRes.slice(5));
-
-			const idMapMock = {};
-
-			nodeRes.slice(1, nodeRes.length - 1).forEach(({ _id }) => { idMapMock[_id] = [_id]; });
-
-			FilesManager.getFile.mockResolvedValueOnce(JSON.stringify(idMapMock));
-
-			const expectedData = cloneDeep(groupData);
-			expectedData.objects = [{
-				container: model,
-				_ids: nodeRes.slice(1, 5).map(({ _id }) => _id),
-			}];
-
-			await expect(Groups.getTicketGroupById(teamspace, project, model, revId, ticket, group))
-				.resolves.toEqual(expectedData);
-
-			expect(GroupsModel.getGroupById).toHaveBeenCalledTimes(1);
-			expect(GroupsModel.getGroupById).toHaveBeenCalledWith(teamspace, project, model, ticket, group);
-
-			expect(RevsModel.getLatestRevision).not.toHaveBeenCalled();
-
-			expect(MetaModel.getMetadataByRules).toHaveBeenCalledTimes(1);
-			expect(MetaModel.getMetadataByRules).toHaveBeenCalledWith(teamspace, project, model, revId,
-				groupData.rules, { parents: 1 });
-
-			expect(ScenesModel.getNodesBySharedIds).toHaveBeenCalledTimes(2);
-			expect(ScenesModel.getNodesBySharedIds).toHaveBeenCalledWith(teamspace, project, model,
-				revId, metaRes.flatMap(({ parents }) => parents), { _id: 1 });
+const testUpdateGroup = () => {
+	const teamspace = generateRandomString();
+	const project = generateRandomString();
+	const container = generateRandomString();
+	const ticket = generateRandomString();
+	const groupId = generateRandomString();
+	const author = generateRandomString();
+
+	describe('Update groups', () => {
+		test('should update a group', async () => {
```

Comment:
this should be a smart group test?

Otherwise i'm confused as to why this exists, if it's just a ticket group with no metadata match then it's just `preserve object as it is (no metadata found)` tetst in the loop

Reply:
ah it was just to cover the scenario that data do not update objects

![image](https://github.com/3drepo/3drepo.io/assets/71635617/aa199059-dafc-40d9-bc3d-c5747a08ce6f)

I could just rename the description to 'should update a group without updating objects' ?

Reply:
right, in which case we should be checking convertToExternalIds is not being called.

I know you can't do it here as it's an internal function, but i've had to expose it in #4641 

Leave this one open, i'll add it on that branch